system/lib/libc/musl/src/internal/pthread_impl.h

```diff
@@ -129,19 +141,28 @@ int __libc_sigprocmask(int, const sigset_t *, sigset_t *);
 void __lock(volatile int *);
 void __unmapself(void *, size_t);
 
-int __timedwait(volatile int *, int, clockid_t, const struct timespec *, void (*)(void *), void *, int);
-void __wait(volatile int *, volatile int *, int, int);
+void __vm_wait(void);
+void __vm_lock(void);
+void __vm_unlock(void);
 
+int __timedwait(volatile int *, int, clockid_t, const struct timespec *, int);
+int __timedwait_cp(volatile int *, int, clockid_t, const struct timespec *, int);
+void __wait(volatile int *, volatile int *, int, int);
+static inline void __wake(volatile void *addr, int cnt, int priv)
+{
+	if (priv) priv = 128;
+	if (cnt<0) cnt = INT_MAX;
 #ifdef __EMSCRIPTEN__
-#define __wake(addr, cnt, priv) emscripten_futex_wake((void*)addr, (cnt)<0?INT_MAX:(cnt))
+	emscripten_futex_wake((void*)addr, (cnt)<0?INT_MAX:(cnt));
```

Comment:
I see you changed the signature of `emscripten_futex_wake` to take in a `volatile void*` instead of just a `void*`. Doesn't that cause this to give a build error, if anything calls `__wake`, because here the cast is still to `void*`. Should cast to `volatile void *` then?

Btw, what was the rationale to change to `volatile void*`?